nix/dapp.nix

```diff
@@ -0,0 +1,740 @@
+# This file has been generated by dapp2nix 0.0.2, do not edit.
+{ solidityPackage, solc, dapp2 }:
+
+let
+  inherit (builtins) map listToAttrs attrNames attrValues;
+  mapAttrs = if (builtins ? mapAttrs)
+    then builtins.mapAttrs
+    else f: attrs:
+      listToAttrs (map
+        (name: { inherit name; value = f name attrs."${name}"; })
+        (attrNames attrs));
+
+  defaults = {
+    inherit solc;
+    test-hevm = dapp2.test-hevm;
+    doCheck = true;
+  };
+
+  package = spec: let
+    spec' = defaults // (removeAttrs spec [ "repo'" "src'" ]);
+  in solidityPackage (spec' // {
+    deps = map (spec:
+      package (spec // { inherit (spec') solc test-hevm doCheck; })
+    ) (attrValues spec'.deps);
+  });
+
+  packageSpecs = mapAttrs (_: package);
```

Comment:
How does this work? If it's supposed to run `package` on each element in an attr-set, shouldn't it be `mapAttrs(spec: package spec)`?

Reply:
Consider `packageSpecs = specs: mapAttrs (_: value: package value) specs;`, here we can remove the trailing variables and function arguments.

Reply:
But how does this work? i.e. how does it know it should apply `package` on the second value passed to `f`?

Reply:
`specs: mapAttrs (_: value: package value) specs` => `mapAttrs (_: value: package value)`
`value: package value` => `package`
`mapAttrs (_: value: package value)` => `mapAttrs (_: package)`